src/panel-app/useImportMapOverrides.js

```diff
@@ -0,0 +1,62 @@
+import { useState, useEffect } from "react";
+import { evalCmd } from "../inspected-window.helper.js";
+
+export default function useImportMapOverrides() {
+  const [importMapsEnabled, setImportMapEnabled] = useState(false);
+  const [overrides, setOverrides] = useState({});
+
+  async function checkImportMapOverrides() {
+    const hasImportMapsEnabled = await evalCmd(`(function() {
+      return !!window.importMapOverrides
+    })()`);
+    setImportMapEnabled(hasImportMapsEnabled);
+  }
+
+  async function getImportMapOverrides() {
+    const { imports } = await evalCmd(`(function() {
+      return window.importMapOverrides.getOverrideMap()
+    })()`);
+    setOverrides(imports);
+  }
+
+  async function addOverride(currentMap, currentUrl) {
+    await evalCmd(`(function() {
+      return window.importMapOverrides.addOverride("${currentMap}", "${currentUrl}")
+    })()`);
+  }
+
+  async function removeOverride(currentMap) {
+    await evalCmd(`(function() {
+      return window.importMapOverrides.removeOverride("${currentMap}")
+    })()`);
+  }
+
+  async function batchSetOverrides() {
+    const overrideCalls = Object.entries(overrides).map(([map, url]) =>
+      !url ? removeOverride(map) : addOverride(map, url)
+    );
+    await Promise.all(overrideCalls);
+    await evalCmd(`window.location.reload()`);
+  }
+
+  // Get initial list of maps if they exist
+  useEffect(() => {
+    const hasImportMapsEnabled = checkImportMapOverrides();
```

Comment:
`checkImportMapOverrdides` returns a promise (since it's an async function), right? so we either need to `await` it or `.then` it...?

Reply:
Oh yea, good catch on this one!

Reply:
Fixed.